src/main/java/uk/co/onsdigital/discovery/metadata/api/controller/MetadataController.java

```diff
@@ -202,12 +203,27 @@ public PlatformTransactionManager getTransactionManager(final EntityManagerFacto
         return new JpaTransactionManager(emf);
     }
 
+    @ExceptionHandler(DataSetNotFoundException.class)
+    void handleDataSetNotFoundException(DataSetNotFoundException e, HttpServletResponse response) throws IOException {
+        logger.error(e.getMessage(), e);
+        response.sendError(HttpStatus.NOT_FOUND.value(), e.getMessage());
+    }
+
+    @ExceptionHandler(DataResourceNotFoundExcecption.class)
+    void handleDataResourceNotFoundException(DataResourceNotFoundExcecption e, HttpServletResponse response) throws IOException {
+        logger.error(e.getMessage(), e);
+        response.sendError(HttpStatus.NOT_FOUND.value(), e.getMessage());
+    }
+
```

Comment:
We might want to take the same approach as the job creator (JobController.java), and make both exceptions extend a common exception (that includes an http status), then have a single handleCommonException method